### PR TITLE
feat(pnpapi): Add `process.versions.pnp`

### PR DESCRIPTION
**Pull request by ExE-Boss**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46700#issuecomment-673998891> (<https://yarnpkg.com/advanced/pnpapi#processversionspnp>)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/pnpapi/index.d.ts [hunk 1]
```diff
@@ -39,3 +40,11 @@ export function resolveRequest(
 ): string | null;
 
 export function setup(): void;
+
+declare global {
+	namespace NodeJS {
```
Comment (danger-public) [inline types/pnpapi/index.d.ts:45]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/pnpapi/index.d.ts [hunk 2]
```diff
@@ -39,3 +40,11 @@ export function resolveRequest(
 ): string | null;
 
 export function setup(): void;
+
+declare global {
+    namespace NodeJS {
+        interface ProcessVersions {
+            pnp: string;
```
Comment (SimenB) [inline types/pnpapi/index.d.ts:47]: Shouldn't it still be optional? Just loading the types doesn't mean it's pnp at runtime

**Comment by amcasey** [comment]
Closing as abandoned.

**Comment by ExE-Boss** [comment]
@amcasey

This is blocked on review from @arcanis.

File: types/pnpapi/index.d.ts (hunk 2 above)
Comment (arcanis) [reply types/pnpapi/index.d.ts:47]: Agreed 👍

**Comment by arcanis** [comment]
Sorry, I didn't see it before! @SimenB's comment should be addressed, then it's fine by me 😃

**Comment by amcasey** [comment]
@ExE-Boss My mistake.  The comment from 30 days ago was unaddressed, so I assumed this was no longer interesting.

**Review by amcasey — CHANGES_REQUESTED** [review]
Please address the feedback from @SimenB.

File: types/pnpapi/pnpapi-tests.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/// <reference types="node"/>
```
Comment (ExE-Boss) [inline types/pnpapi/pnpapi-tests.ts:1]: I’ve added `@types/node` to `devDependencies` so that changes to **Node** type definitions won’t cause breakage like with `@types/webpack‑env`: <https://github.com/DefinitelyTyped/DefinitelyTyped/issues/47602> (PR <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/47516>).

**Review by amcasey — APPROVED** [review]

**Comment by ExE-Boss** [comment]
Ready to merge

**Pull request merged**